src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/CanonicalSearchValue.cs

```diff
@@ -0,0 +1,133 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text.RegularExpressions;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.SearchValues
+{
+    /// <summary>
+    /// Represents an URI search value.
+    /// </summary>
+    public class CanonicalSearchValue : UriSearchValue
```

Comment:
Why are we creating a derived type for this? We group by search value type for inserting in SQL, so I can see this approach leading to some problems.

Reply:
I this what the mapping is used for?

https://github.com/microsoft/fhir-server/blob/e476c7ba17d92eb9f7118f2790c6cae284331351/src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SearchParameterToSearchValueTypeMap.cs#L50-L53

Reply:
Yes. Elsewhere in there, though, you can see the assumption that `SearchParamType.Uri => typeof(UriSearchValue)`. I guess it's not technically a problem, but I'm still wondering why we are introducing this derived type instead of adding fields. 

Also, should we only interpret `|` as a version separator if the field datatype is canonical?

Reply:
Moved to using fields